### ISSUE TITLE
Join Group Error

### ISSUE THREAD

**Issue by cakyel** [swe574-2015fall-group2/android_repo#40]
When I tried to join Banana group via All Groups screen, the application crashed. Details are below.

![image](https://cloud.githubusercontent.com/assets/4705964/11919685/faac88f8-a761-11e5-94aa-3eaf56af7bdd.png)

12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime: Process: swe574.boun.edu.androidproject, PID: 13891
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime: java.lang.RuntimeException: An error occured while executing doInBackground()
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at android.os.AsyncTask$3.done(AsyncTask.java:300)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.FutureTask.run(FutureTask.java:242)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.lang.Thread.run(Thread.java:841)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:  Caused by: java.lang.IllegalStateException: Response code is not valid
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at swe574.boun.edu.androidproject.tasks.ApplyGroupTask.doInBackground(ApplyGroupTask.java:79)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at swe574.boun.edu.androidproject.tasks.ApplyGroupTask.doInBackground(ApplyGroupTask.java:27)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at android.os.AsyncTask$2.call(AsyncTask.java:288)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231) 
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112) 
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587) 
12-20 14:37:02.871 13891-13932/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.lang.Thread.run(Thread.java:841) 
12-20 14:37:03.761 13891-13891/swe574.boun.edu.androidproject W/EGL_emulation: eglSurfaceAttrib not implemented
12-20 14:38:27.391 13891-13932/swe574.boun.edu.androidproject I/Process: Sending signal. PID: 13891 SIG: 9

**Comment by cakyel** [comment]
By the way now I see that I have joined the group even if it was crashed.

**Comment by OrkunKocyigit** [comment]
It is cause backend sends as we fucked it up response code while doing everything right on their back side >_>

**Comment by OrkunKocyigit** [comment]
I cannot reproduce this problem, I added extra debug info to our logcat. If you can try to sync and reproduce the problem and repost LogCat. Other I will close this issue in 7 days.

**Comment by cakyel** [comment]
I think I misinterpreted the problem. I tried it again and figured out the problem below.
- First I joined the banana group.
- After I joined, I tapped the group again and it asked again to me to join the same group. After that, the app crashed.

![image](https://cloud.githubusercontent.com/assets/4705964/11919900/5d82e6c0-a767-11e5-93c4-62faeb858a6b.png)

12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #1
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime: Process: swe574.boun.edu.androidproject, PID: 1602
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime: java.lang.RuntimeException: An error occured while executing doInBackground()
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at android.os.AsyncTask$3.done(AsyncTask.java:300)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.FutureTask.run(FutureTask.java:242)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.lang.Thread.run(Thread.java:841)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:  Caused by: java.lang.IllegalStateException: Response code is not valid 500 RESPOnSE JSON 
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime: {"status":"error","consumerMessage":"User is already member of this group","applicationMessage":""}
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at swe574.boun.edu.androidproject.tasks.ApplyGroupTask.doInBackground(ApplyGroupTask.java:84)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at swe574.boun.edu.androidproject.tasks.ApplyGroupTask.doInBackground(ApplyGroupTask.java:27)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at android.os.AsyncTask$2.call(AsyncTask.java:288)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
12-20 15:15:08.775 1602-1638/swe574.boun.edu.androidproject E/AndroidRuntime:       ... 4 more
12-20 15:15:09.175 1602-1602/swe574.boun.edu.androidproject W/EGL_emulation: eglSurfaceAttrib not implemented
12-20 15:15:10.755 1602-1602/swe574.boun.edu.androidproject D/dalvikvm: GC_FOR_ALLOC freed 653K, 13% free 5374K/6148K, paused 46ms, total 48ms
